### ISSUE TITLE
Multiple Sync Agents / User mismatch for client file identifier [IDENT]

### ISSUE THREAD

**Issue by davidcockburngithub** [realm/realm-swift#8117]
I also have this issue - I have downloaded the 'Template App' available from App Services (Realm) for Swift/SwiftUI - Todo, with Flex Sync. This app (which is entirely written by Realm/MongoDB) re-creates the exact behaviour above, whereby there is an issue when multiple users log in/out/back-in using a single device, straight out of the box. User 1 login is successful, then User 1 logout is successful. User 2 login then initially appears successful, before displaying a Progress View, and then displays an error on screen 'Automatic Client Reset Failed' [User mismatch for client file identifier (IDENT)] and the log shows a Client File Ident Error (Multiple Sync Agents)? Can anyone from MongoDB assist or explain this? There don't appear to be any errors in the code? I suspect that this is either a session error, or that User 1 is not being fully 'dropped' prior to User 2 trying to login?

Logs in App Services show that the initial connection by User 2 was OK, but then subsequently failed with the 'Client File Ident Error - code 223)?

I have copied the code below, but it's exactly as is available from: https://www.mongodb.com/docs/atlas/app-services/reference/template-apps/#std-label-template-apps

ContentView [initial]:

```
import SwiftUI
import RealmSwift

struct ContentView: View {
    @ObservedObject var app: RealmSwift.App
    @EnvironmentObject var errorHandler: ErrorHandler

    var body: some View {
        if let user = app.currentUser {
            // Setup configuraton so user initially subscribes to their own tasks
            let config = user.flexibleSyncConfiguration(initialSubscriptions: { subs in
                subs.remove(named: Constants.allItems)
                if let _ = subs.first(named: Constants.myItems) {
                    // Existing subscription found - do nothing
                    return
                } else {
                    // No subscription - create it
                    subs.append(QuerySubscription<Item>(name: Constants.myItems) {
                        $0.owner_id == user.id
                    })
                }
            })
            OpenRealmView(user: user)
                // Store configuration in the environment to be opened in next view
                .environment(\.realmConfiguration, config)
        } else {
            // If there is no user logged in, show the login view.
            LoginView()
        }
    }
}
```

OpenRealmView [subsequent view]:

```
import SwiftUI
import RealmSwift

/// Called when login completes. Opens the realm asynchronously and navigates to the Items screen.
struct OpenRealmView: View {
    @AsyncOpen(appId: theAppConfig.appId, timeout: 2000) var asyncOpen
    // We must pass the user, so we can set the user.id when we create Item objects
    @State var user: User
    @State var showMyItems = true
    @State var isInOfflineMode = false
    // Configuration used to open the realm.
    @Environment(\.realmConfiguration) private var config

    var body: some View {

        switch asyncOpen {
        case .connecting:
            // Starting the Realm.asyncOpen process.
            // Show a progress view.
            ProgressView()
        case .waitingForUser:
            // Waiting for a user to be logged in before executing
            // Realm.asyncOpen.
            ProgressView("Waiting for user to log in...")
        case .open(let realm):
            // The realm has been opened and is ready for use.
            // Show the Items view.
            ItemsView(leadingBarButton: AnyView(LogoutButton()), user: user, showMyItems: $showMyItems, isInOfflineMode: $isInOfflineMode)
                // showMyItems toggles the creation of a subscription
                // When it's toggled on, only the original subscription is shown -- "my_items".
                // When it's toggled off, *all* items are downloaded to the
                // client, including from other users.
                .onChange(of: showMyItems) { newValue in
                    let subs = realm.subscriptions
                    subs.update {
                        if newValue {
                            subs.remove(named: Constants.allItems)
                        } else {
                            if subs.first(named: Constants.allItems) == nil {
                                subs.append(QuerySubscription<Item>(name: Constants.allItems))
                            }
                        }
                    }
                }
                // isInOfflineMode simulates a situation with no internet connection.
                // While sync is not available, items can still be written and queried.
                // When sync is resumed, items created or updated offline will upload to
                // the server and changes from the server or other devices will be downloaded to the client.
                .onChange(of: isInOfflineMode) { newValue in
                    let syncSession = realm.syncSession!
                    newValue ? syncSession.suspend() : syncSession.resume()
                }
                .onAppear {
                    if let _ = realm.subscriptions.first(named: Constants.allItems) {
                        // The client was subscribed to all items from a previous
                        // session, so set UI toggle accordingly
                        showMyItems = false
                    }
                }
        case .progress(let progress):
            // The realm is currently being downloaded from the server.
            // Show a progress view.
            ProgressView(progress)
        case .error(let error):
            // Opening the Realm failed.
            // Show an error view.
            ErrorView(error: error)
        }
    }
}
```

_Originally posted by @davidcockburngithub in https://github.com/realm/realm-swift/issues/8069#issuecomment-1414519792_

**Comment by ericjordanmossman** [comment]
@davidcockburngithub I'm seeing the failed automatic recovery error also. Followed by a loop of failed retries. I get this with realm-swift 10.32.3. Could you set the realm-swift version to 10.34.1? Then File > Packages > Reset Package Caches. I believe the reported issue may be resolved in 10.34.0. I no longer see the issue in starting in that version.
<img width="785" alt="Screenshot 2023-02-03 at 10 38 36 AM" src="https://user-images.githubusercontent.com/25092858/216645127-66a7e613-a43d-434a-bab3-bca63da78fb3.png">

I'll update the template app version, if this is the case.

**Comment by davidcockburngithub** [comment]
@ericjordanmossman you are correct! This has indeed resolved the problem - I have switched between authenticated users multiple times, and the issue no longer appears - I think updating the template would be a good idea :) Thank you for your quick reply - as a secondary issue, using the latest version (10.34.1) this now appears in the Xcode log (but it is not causing any obvious problems, as yet) - just reporting in-case you are not aware.

**Thread Performance Checker: Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions**
